### DISCUSSION TITLE
Inquiry: Triggering State Update Based on Home Assistant Event

### DISCUSSION THREAD

**Discussion by michaelkun** [guerrerotook/securitas-direct-new-api#236] [General]
Hello,

I am wondering if there is an existing method to trigger a state update in Home Assistant based on an event. My goal is to disable the continuous state polling loop and instead query the current state only when it is needed for an automation.

I have noticed that there is a service called **refresh_alarm_status** in Home Assistant, but invoking it doesn't seem to produce any results. Is there a way to force a state query from Home Assistant at a given moment, thereby avoiding continuous polling and querying the state only when required?

**Thank you for your assistance.**

**Comment by bigramonk** [comment]
Agree, a working _securitas.refresh_alarm_status_ service would be great.
I tried to use it upon normal events that could trigger a normal status change such as entrance/garage door opening/closing for example but it does not seem to work as well as expected.
Should this work, an in-between call every hour would be more than enough for me.
